src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TupleCreationExpression.cs

```diff
@@ -29,9 +29,14 @@ private BoundNode VisitTupleExpression(BoundTupleExpression node)
         /// For instance, for a long tuple we'll generate:
         /// creationExpression(ctor=largestCtor, args=firstArgs+(nested creationExpression for remainder, with smaller ctor and next few args))
         /// </summary>
-        private BoundNode RewriteTupleCreationExpression(BoundTupleExpression node, ImmutableArray<BoundExpression> rewrittenArguments)
+        private BoundExpression RewriteTupleCreationExpression(BoundTupleExpression node, ImmutableArray<BoundExpression> rewrittenArguments)
         {
-            NamedTypeSymbol underlyingTupleType = node.Type.TupleUnderlyingType;
+            return MakeTupleCreationExpression(node.Syntax, (TupleTypeSymbol)node.Type, rewrittenArguments) ?? node;
+        }
+
+        private BoundExpression MakeTupleCreationExpression(CSharpSyntaxNode syntax, TupleTypeSymbol type, ImmutableArray<BoundExpression> rewrittenArguments)
+        {
+            NamedTypeSymbol underlyingTupleType = type.TupleUnderlyingType;
```

Comment:
This code is in lowering. I think we can only have real tuples here, not underlying types.
I will look at other similar cases.